### DISCUSSION TITLE
Any idiomatic way I could have pull requests run a template post update command?

### DISCUSSION THREAD

**Discussion by sheldonhull** [renovatebot/renovate#29579] [Request Help]
I use a tool that creates a change log fragment file for updates. The dependency updates I'd like to include in this. This allows me to work on trunk and I'm ready to release all of these change entries get matched up into an updated changelog file. 

I'm not certain if this is possible with renovate. As it creates a PR I'd really like to have the list of dependencies that were updated either run a for each with a commandline tool that logs the dependency that was updated or worst case just a single string that I could pass into append. Since it powers our semantic versioning it would be great to include this. 

https://changie.dev/guide/quick_start/

Worst case I'll just create another CLI integration that creates the entry after parsing pull request that was just merged. Figured I'd check though since I'm it clear if the post upgrade scripts or tasks might solve this. If I run outside of renovate the pull request will stop being updated due to being changed outside of renovate. 

If this logic for saying "updated outside renovate is just matching the git commit author maybe I just run command outside of renovate with its username and problem solved.

**Comment by rarkins** [comment]
Try postUpgradeTasks

**Reply by sheldonhull** [reply]
Ok, i'll report back for others benefit if this works since seems possible. I'll have to find the available template fields it has and take a peek. Cheers!

**Reply by sheldonhull** [reply]
Ok! I got it working so thanks for confirming. A follow-up question. 

I had to add in my global `config.js` (this is fine because I control all the repos, I'm fine with any post upgrade commands being allowed. 


```
  allowedPostUpgradeCommands: ['.*'],
  allowPostUpgradeCommandTemplating: true,
```

```json
  "postUpgradeTasks": {
    "commands": [
      "docker pull ghcr.io/miniscruff/changie:latest 2> /dev/null",
      "docker run -v $PWD:/src -w /src --user $(id -u ${USER}):$(id -g ${USER}) ghcr.io/miniscruff/changie:latest new --kind 'dependencies' --body \"Renovate update {{updateType}}: {{datasource}} {{ depName }}.\""
    ],
    "fileFilters": [".changes/unreleased/*.yaml"],
    "executionMode": "update"
  }
```

This ran and created a single entry. 

- If I have a PR that is grouping, will it run for each individual dependency being updated, or once for the entire group?
- If the group runs just once, is there a friendly template variable I can use that would give me the specific dependency name and version to and from in so each entry gets an entry or not? 

So right now the entry it created was useful but I'd like to be more more specific if I can so the dependency log includes each dependency upgrade detail so the final changelog provides full context.

**Reply by rarkins** [reply]
Look at executionMode: https://docs.renovatebot.com/configuration-options/#executionmode

By default it's run once per update but you can switch to branch.
